android/src/main/java/com/github/wumke/RNImmediatePhoneCall/RNImmediatePhoneCallModule.java

```diff
@@ -34,35 +35,59 @@ public String getName() {
     }
 
     @ReactMethod
-    public void immediatePhoneCall(String number) {
-        RNImmediatePhoneCallModule.number = Uri.encode(number);
+    public void immediatePhoneCall(String number, Promise promise) {
 
-        if (ContextCompat.checkSelfPermission(reactContext.getApplicationContext(),
-                android.Manifest.permission.CALL_PHONE) == PackageManager.PERMISSION_GRANTED) {
-            call();
-        } else {
-            ActivityCompat.requestPermissions(getCurrentActivity(),
+        if (number == null || number.isEmpty()) {
+            promise.reject(new JSApplicationIllegalArgumentException("Invalid number: " + url));
+            return;
+        }
+
+        try {
+            Bool hasPermission = ContextCompat.checkSelfPermission(reactContext.getApplicationContext(),
```

Comment:
Dont think there is Bool in java, should be Boolean. https://docs.oracle.com/javase/8/docs/api/index.html?java/lang/Boolean.html

Reply:
Unless this is some custom object?